woodwork/deserialize.py

```diff
@@ -135,19 +113,8 @@ def _typing_information_to_woodwork_table(table_typing_info, validate, **kwargs)
     elif load_format == "orc":
         dataframe = lib.read_orc(file)
 
-    dataframe.ww.init(
-        name=table_typing_info.get("name"),
-        index=table_typing_info.get("index"),
-        time_index=table_typing_info.get("time_index"),
-        logical_types=logical_types,
-        semantic_tags=semantic_tags,
-        use_standard_tags=use_standard_tags,
-        table_metadata=table_typing_info.get("table_metadata"),
-        column_metadata=column_metadata,
-        column_descriptions=column_descriptions,
-        column_origins=column_origins,
-        validate=validate,
-    )
+    init_params = _typing_info_to_init_params(table_typing_info)
```

Comment:
As I have been going through this update, I have also come to a similar conclusion that our entire serialization/deserialization approach might benefit from some refactoring. Basically, what led me to this conclusion, is that it's really not possible to change the way we serialize to parquet files inside of our `ww.df.to_disk` method, without changing a lot of other helper functions that are also used for serializing to other file types.

To me, it feels like we should be able to change the way we serialize to parquet without potentially impacting/breaking the way we serialize to other formats - possibly by just updating a `ParquetSerializer` class or something like that. The fact that we can't do that right now leads me to believe there might be a better design for our serialization code.

I suggest we give things a bit more thought before we merge in this PR and decide how we want to proceed.

Reply:
Yeah, interesting thoughts.  And actually what you describe is I think a classic use-case for OO programming.  You have a method that does some piece of a process or algorithm and then you want to have another thing like that but just have it do something slightly different.  So you make a subclass of the class that contains the method and override it with your extra behavior.  That could help with the situation you describe where it's hard to change one helper function without impacting a lot of other things.  Having that helper function in a class would allow changing it in a subclass without impacting other things that depend on the base class.